### PR TITLE
Fix directory and file detection on Windows

### PR DESCRIPTION
**Pull request by ihhub**
It saves a bit of time.

### CODE REVIEW

**Review by oleg-derevenetz — COMMENTED** [review]
Hi @ihhub There are also `System::IsFile()` and `System::IsDirectory()`. While with POSIX both of them work correctly (by checking the inode type), it seems that with Windows and other systems they do not check the file type at all. We can fix them and use here. What do you think?

**Comment by ihhub** [comment]
> Hi @ihhub There are also `System::IsFile()` and `System::IsDirectory()`. While with POSIX both of them work correctly (by checking the inode type), it seems that with Windows and other systems they do not check the file type at all. We can fix them and use here. What do you think?

Sure. We can do this.

**Review by oleg-derevenetz — APPROVED** [review]

File: src/engine/system.cpp [hunk 1]
```diff
@@ -223,7 +223,23 @@ char * System::GetOptionsArgument( void )
...
+    if ( fileAttributes == INVALID_FILE_ATTRIBUTES ) {
+        // This path doesn't exist.
+        return false;
+    }
+
+    if ( ( fileAttributes & FILE_ATTRIBUTE_DIRECTORY ) != 0 ) {
+        // This is a directory.
+        return false;
+    }
+
```
Comment (oleg-derevenetz) [inline src/engine/system.cpp:242]: BTW we can add TODO comments for `ANDROID` and `FHEROES2_VITA` cases to not forget to add the similar checks in future here and in the `System::IsDirectory()`.

Comment (ihhub) [reply src/engine/system.cpp:242]: It is done.

**Pull request merged**